compiler/src/main/scala/org/scalajs/core/compiler/PrepJSInterop.scala

```diff
@@ -713,3 +819,46 @@ abstract class PrepJSInterop extends plugins.PluginComponent
     }
 
 }
+
+object PrepJSInterop {
+  private final class OwnerKind(val baseKinds: Int) extends AnyVal {
+    import OwnerKind._
+
+    @inline def isBaseKind: Boolean =
+      Integer.lowestOneBit(baseKinds) == baseKinds && baseKinds != 0 // exactly 1 bit on
+
+    @inline def |(that: OwnerKind): OwnerKind =
+      new OwnerKind(this.baseKinds | that.baseKinds)
+
+    @inline def is(that: OwnerKind): Boolean =
+      (this.baseKinds & that.baseKinds) != 0
+
+    @inline def isnt(that: OwnerKind): Boolean =
+      !this.is(that)
+  }
+
+  private object OwnerKind {
+    val None = new OwnerKind(0x00)
+
+    val NonEnumScalaClass = new OwnerKind(0x01)
+    val NonEnumScalaMod = new OwnerKind(0x02)
+    val JSNativeClass = new OwnerKind(0x04)
+    val JSNativeMod = new OwnerKind(0x08)
+    val JSClass = new OwnerKind(0x10)
+    val EnumClass = new OwnerKind(0x20)
+    val EnumMod = new OwnerKind(0x40)
+    val EnumImpl = new OwnerKind(0x80)
```

Comment:
Please add short doc comments to the three enum constants. It is hard to distinguish what they mean without looking at how they are used.